mobilecoinfoundation/attestation/grpcio/build.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Generate the grpcio code for the gRPC service.
+
+use std::fs;
+use std::path::{Path, PathBuf};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let out_dir = std::env::var("OUT_DIR")?;
+    grpcio_compiler::prost_codegen::compile_protos(
+        &["../v1/services.proto"],
+        &["../../../", "../"],
+        &out_dir,
+    )?;
+
+    // This is not ideal and is likely to be fragile longer term.
+    // Ideally there would be a `protoc` plugin like
+    // https://crates.io/crates/protoc-gen-tonic for generating just the
+    // grpcio services.
+    let out_dir = PathBuf::from(out_dir).join("mobilecoinfoundation.attestation.v1.rs");
+    remove_messages_from_grpcio_services(out_dir)?;
+    Ok(())
+}
+
+/// Removes the message types from the generated grpcio rust file.
+///
+/// # Arguments:
+/// * `path` - The path to the generated grpcio file.
+fn remove_messages_from_grpcio_services(
+    path: impl AsRef<Path>,
```

Comment:
Do we expect this method to take anything other than a `Cow`? I think it's fine to leave as is if not, but just curious.

Reply:
I've been leaning on `impl AsRef<>` for the [common implementations](https://doc.rust-lang.org/std/convert/trait.AsRef.html#implementors), any `pathlike`, `stringlike`, or `[u8]` like items.
I might be overusing it, but it seems easy enough to use, to provide a more ergonomic api, even if that api is only used in one place.